code/modules/mob/living/simple_animal/hostile/megafauna/colossus.dm

```diff
@@ -359,4 +359,407 @@ Difficulty: Very Hard
 /obj/machinery/smartfridge/black_box/default_deconstruction_crowbar()
 	return
 
+///Anomolous Crystal///
+
+/obj/machinery/anomalous_crystal
+	name = "anomalous crystal"
+	desc = "A strange chunk of crystal, being in the presence of it fills you with equal parts excitement and dread."
+	icon = 'icons/obj/lavaland/artefacts.dmi'
+	icon_state = "anomaly_crystal"
+	luminosity = 8
+	burn_state = LAVA_PROOF
+	pixel_y = -4
+	use_power = 0
+	density = 1
+	languages_spoken = ALL
+	languages_understood = ALL
+	flags = HEAR
+	var/activation_turf = null
+	var/aim_direction = SOUTH
+	var/activation_method = "touch"
+	var/activation_damage_type = null
+	var/last_use_timer = 0
+	var/cooldown_add = 30
+	var/list/affected_targets = list()
+	var/activation_sound = 'sound/effects/break_stone.ogg'
+
+/obj/machinery/anomalous_crystal/New()
+	activation_method = pick("touch","beam","bullet","energy","magic","mob_bump","heat","weapon","speech")
+	..()
+
+/obj/machinery/anomalous_crystal/Hear(message, atom/movable/speaker, message_langs, raw_message, radio_freq, spans)
+	..()
+	if(isliving(speaker))
+		ActivationReaction(speaker,"speech")
+
+/obj/machinery/anomalous_crystal/attack_hand(mob/user)
+	..()
+	ActivationReaction(user,"touch")
+
+/obj/machinery/anomalous_crystal/attackby(obj/item/I, mob/user, params)
+	if(I.is_hot())
+		ActivationReaction(user,"heat")
+	else
+		ActivationReaction(user,"weapon")
+	..()
+
+/obj/machinery/anomalous_crystal/bullet_act(obj/item/projectile/P, def_zone)
+	..()
+	if(istype(P, /obj/item/projectile/beam))
+		ActivationReaction(P.firer, "beam", P.damage_type)
+		return
+	if(istype(P, /obj/item/projectile/bullet))
+		ActivationReaction(P.firer, "bullet", P.damage_type)
+		return
+	if(istype(P, /obj/item/projectile/energy))
+		ActivationReaction(P.firer, "energy",P.damage_type)
+		return
+	if(istype(P, /obj/item/projectile/magic))
+		ActivationReaction(P.firer, "magic", P.damage_type)
+		return
```

Comment:
Instead of all of these istypes, do `ActivationReaction(P.firer, P.flag, P.damage_type)` which also happens to give you 'bomb' as a possible unlock require
**important note; you'll have to change any references to "beam" for ActivationReaction() to "laser"** but at the moment that's 1 thing, line 387